src/random/distributions/GenerateCanonical.i.hh

```diff
@@ -12,35 +12,34 @@ namespace celeritas
 /*!
  * Generate random numbers in [0, 1).
  */
-template<class Generator, class T>
-CELER_FUNCTION auto GenerateCanonical<Generator, T>::operator()(Generator& rng)
+template<class Generator, class RealType>
+CELER_FUNCTION auto
+GenerateCanonical<Generator, RealType>::operator()(Generator& rng)
     -> result_type
 {
-    return std::generate_canonical<result_type,
-                                   std::numeric_limits<result_type>::digits>(
-        rng);
+    using limits_t = std::numeric_limits<result_type>;
+    return std::generate_canonical<result_type, limits_t::digits>(rng);
 }
 
-#ifdef __NVCC__
 //---------------------------------------------------------------------------//
 /*!
- * Specialization for RngEngine, float
+ * Helper function to generate a random real number in [0, 1).
  */
-__device__ float GenerateCanonical<RngEngine, float>::operator()(RngEngine& rng)
+template<class RealType, class Generator>
+CELER_FUNCTION RealType sample_uniform(Generator& g)
```

Comment:
```suggestion
CELER_FUNCTION RealType generate_canonical(Generator& g)
```

Reply:
Uh oh, if this is still here how is everything building and passing? 😱 Good catch!